Shopify/hydrogen
# Fd login command

Closes https://github.com/Shopify/core-issues/issues/56397

### WHY are these changes introduced?

We are refactoring how CLI login works. For that, we are introducing `h2 login` and `h2 logout`, as well as removing `--shop` flags from certain commands.

### WHAT is this pull request doing?

- Implements `h2 login` and `h2 logout` commands.
- Removes `--shop` flag from other commands.
- Given the required changes for the previous 2 features, I've refactored some modules in the CLI related with auth and admin requests, and added some extra tests.

### HOW to test your changes?

Play with different permutations of the related commands:
- h2 logout
- h2 login
- h2 link
- h2 unlink
- h2 env list
- h2 env pull

When not logged in, they should prompt to login. When not linked, they should ask to run `h2 link` (env list/pull).

💡  I think it would be better to check the code commit by commit. 

❓ Should we mark this as a major bump for the CLI package? It's deprecating flags we just introduced in the last release.

#### Checklist
